kustomize/internal/commands/edit/add/flagsandargs.go

```diff
@@ -22,6 +22,8 @@ type flagsAndArgs struct {
 	// EnvFileSource to derive the configMap/Secret from (optional)
 	// TODO: Rationalize this name with Generic.EnvSource
 	EnvFileSource string
+	// Resource generation behavior (optional)
+	Behavior string
```

Comment:
Please use
https://github.com/kubernetes-sigs/kustomize/blob/master/api/types/generationbehavior.go
 instead of a raw string here.

Reply:
Accept.